modules/registrars/openprovider/Controllers/Hooks/DomainLockingEnabledController.php

```diff
@@ -0,0 +1,154 @@
+                });
+                document.addEventListener("DOMContentLoaded", function() {
+                    var alerts = document.querySelectorAll(\'.alert.alert-danger\');
+                    alerts.forEach(function(alert) {
+                        var strongTag = alert.querySelector(\'strong\');
+                        if (strongTag && strongTag.textContent.trim() === \'Domain Currently Unlocked!\') {
+                            alert.remove();
+                        }
+                    });
+                });
+            </script>';
+    }
+
+    public function handleDomainLockingClientSidebar(\WHMCS\View\Menu\Item $primarySidebar)
+    {
+        $id = $_REQUEST['id'] ?? ($_REQUEST['domainid'] ?? null);
+        if (!$id) {
+            return "";
+        }
+        // Get the domain details
+        $domain = $this->domain->find($id);
+        // Check if OpenProvider is the provider
+        if (!$domain || $domain->registrar !== "openprovider") {
+            return "";
+        }
+
+        if ($this->isDomainLockingEnabled($domain)) {
+            return "";
+        }
+
+        if (!is_null($primarySidebar->getChild('Domain Details Management'))) {
+            $primarySidebar->getChild('Domain Details Management')->removeChild('Registrar Lock Status');
+        }
+    }
+
+    public function handleDomainLockingAdminArea()
+    {
+        if (basename($_SERVER['PHP_SELF'] ?? (($_SERVER['SCRIPT_NAME'] ?? $_SERVER['SCRIPT_FILENAME']) ?? '')) !== "clientsdomains.php") {
+            return "";
+        }
+        $id = $_REQUEST['id'] ?? ($_REQUEST['domainid'] ?? null);
+        if (!$id) {
+            return "";
+        }
+        // Get the domain details
+        $domain = $this->domain->find($id);
+        // Check if OpenProvider is the provider
+        if (!$domain || $domain->registrar !== "openprovider") {
+            return "";
+        }
+
+        if ($this->isDomainLockingEnabled($domain)) {
+            return "";
+        }
+
+        return '<script>
+            $(function (){
+                $(\'td.fieldarea input[type="checkbox"][name="lockstatus"]\').closest(\'tr\').remove();
+            });
+        </script>';
+    }
+
+    private function isDomainLockingEnabled($domain)
+    {
+        try {
+            $op_domain_obj = DomainFullNameToDomainObject::convert($domain->domain);
+            $lockable_state = Cache::get($op_domain_obj->extension);
+
+            if (!isset($lockable_state)) {
+                $op_domain = $this->apiHelper->getDomain($op_domain_obj);
+                if (($op_domain['isLockable'] ?? false)) {
+                    Cache::set($op_domain['domain']['extension'], true);
+                    return true;
+                } else {
+                    Cache::set($op_domain['domain']['extension'], false);
+                    return false;
+                }
+            } else {
+                if ($lockable_state) {
+                    return true;
```

Comment:
even value could save false according to the line 141.
better to return variable itself

Reply:
Yeah. Changed the return value just to lockable_state instead of if condition use